### PR TITLE
Fix bug in how we compute sample clinical data for patient attributes

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

File: src/shared/components/oncoprint/DataUtils.ts [hunk 1]
```diff
@@ -389,9 +410,10 @@ export function makeClinicalTrackData(
         = _.groupBy(data,
         isSampleList(cases) ?
             datum=>datum.uniqueSampleKey :
-            datum=>datum.uniquePatientKey);// TS error will be gone when ersin updates API
+            datum=>datum.uniquePatientKey);
 
     // Create oncoprint data
+    const getDataForCase = makeGetDataForCase(attribute, isSampleList(cases) ? "sample" : "patient", data);
```
Comment (alisman) [inline src/shared/components/oncoprint/DataUtils.ts]: i'd vote for using parens around both the condition and the entire ternary here to make this more readable.

Comment (adamabeshouse) [reply src/shared/components/oncoprint/DataUtils.ts]: around the condition seems excessive... around the ternary I agree. but if you feel strongly I dont care that much

Comment (adamabeshouse) [reply src/shared/components/oncoprint/DataUtils.ts]: ah whatever it does look a little nicer that way

File: src/shared/components/oncoprint/DataUtils.ts [hunk 2]
```diff
@@ -379,6 +379,27 @@ function fillClinicalTrackDatum(
     return trackDatum;
 }
 
+function makeGetDataForCase(
```
Comment (alisman) [inline src/shared/components/oncoprint/DataUtils.ts]: this is sort of crying out for a test.  i have a tool that should make it quite easy.  i'll merge and we can revisit.

Comment (adamabeshouse) [reply src/shared/components/oncoprint/DataUtils.ts]: true, show me the tool when you get a chance

**Pull request merged**